example/calculate_pi/calculate_pi.cpp

```diff
@@ -37,11 +37,50 @@ namespace calculate_pi {
 
 void SetInOrOut(MeshBlockData<Real> *rc) {
   auto pmb = rc->GetBlockPointer();
+
+  ParArrayND<Real> v;
+  const auto &radius = pmb->packages.Get("calculate_pi")->Param<Real>("radius");
+
+  // If we're using sparse variables, we only allocate in_or_out on blocks that have at
+  // least some part inside the circle, otherwise it would be all 0's
+  bool const use_sparse = pmb->packages.Get("calculate_pi")->Param<bool>("use_sparse");
+  if (use_sparse) {
+    int const ndim = pmb->pmy_mesh->ndim;
+    auto &bs = pmb->block_size;
+    // check if block falls on radius.
+    Real coords[4][2] = {{bs.x1min, bs.x2min},
+                         {bs.x1min, bs.x2max},
+                         {bs.x1max, bs.x2min},
+                         {bs.x1max, bs.x2max}};
```

Comment:
Hmm, I'm not sure I follow. I think the check should only check the corners of the interior cells, because we check if the entire block is inside or outside. The ghost zones belong to the neighboring blocks, so they are irrelevant to determine if *this* block is inside or outside.

The if `fully_outside` is false, then we allocate the sparse field on this block and the loop below gets executed. Otherwise, the loop doesn't get executed at all. Since the loop only runs over the current block, I don't see how it can crash. Of course, you are right that we will have a difficulty when trying to communicate boundary ghost zones between a fully outside block and a non-fully outside one, since one has `SetInOrOut` allocated and the other doesn't. But that case will have to be handled by the general true sparse infrastructure.

Does this make sense or did I totally miss what you were saying?

Reply:
Got it. I didn't see that the function returns early.